pkg/tfgen/docs.go

```diff
@@ -47,32 +47,34 @@ import (
 type argumentDocs struct {
 	// The description for this argument.
 	description string
-
-	// (Optional) The names and descriptions for each argument of this argument.
-	arguments map[string]string
-
-	// Whether this argument was derived from a nested object. Used to determine
-	// whether to append descriptions that have continued to the following line
-	isNested bool
 }
 
 // Included for testing convenience.
 func (ad argumentDocs) MarshalJSON() ([]byte, error) {
-	j, err := json.Marshal(struct {
-		Description string
-		Arguments   map[string]string
-		IsNested    bool
+	return json.Marshal(struct {
+		Description string `json:"description"`
 	}{
 		Description: ad.description,
-		Arguments:   ad.arguments,
-		IsNested:    ad.isNested,
 	})
-	if err != nil {
-		return nil, err
-	}
-	return j, nil
 }
 
+type docsPath string
+
+func (d docsPath) nested() bool                 { return strings.Contains(string(d), ".") }
+func (d docsPath) leaf() string                 { s := d.parts(); return s[len(s)-1] }
+func (d docsPath) join(segment string) docsPath { return docsPath(string(d) + "." + segment) }
+func (d docsPath) parts() []string              { return strings.Split(string(d), ".") }
+func (d docsPath) withOutRoot() docsPath        { s := d.parts(); return docsPath(strings.Join(s[1:], ".")) }
+
+type docsPathArr []docsPath
+
+var _ = (sort.Interface)((docsPathArr)(nil))
```

Comment:
I'm guessing this is somehow declaring that docsPathArr implements sort.Interface, but I don't quite follow how it works.